### PR TITLE
Change vext behavior if the bridge is inactive

### PR DESCRIPTION
**Pull request by batconjurer**
* Adds a storage key indicating the active status of the bridge. 
* Adds a quick query method to check if the bridge is active.
* Makes sure the vexts / protocol txs for eth events and bridge pool roots are not issued if the bridge is not active.
* Makes sure the vexts / protocol txs are not accepted if the the bridge is not active.

### CODE REVIEW

File: apps/src/lib/node/ledger/shell/vote_extensions.rs [hunk 1]
```diff
@@ -344,6 +371,104 @@ where
...
+    ) {
+        let mut eth_evs = vec![];
+        let mut bp_roots = vec![];
+        let mut valset_upds = vec![];
+        let bridge_active = self.storage.is_bridge_active();
+
+        for ext in deserialize_vote_extensions(vote_extensions) {
+            if let Some(validator_set_update) = ext.validator_set_update {
+                valset_upds.push(validator_set_update);
+            }
```
Comment (sug0) [inline apps/src/lib/node/ledger/shell/vote_extensions.rs:456]: validator set updates should also be excluded from proposals, but they must be included an epoch before the bridge is started. therefore, `is_bridge_active()` needs an `Epoch` param. we should check if the bridge is active at the next epoch, to include valset upds

Comment (sug0) [reply apps/src/lib/node/ledger/shell/vote_extensions.rs:456]: ah nevermind, just realized we have `check_bridge_status()`. this should be what we should be querying, then

**Review by sug0 — CHANGES_REQUESTED** [review]
hm a couple of things:
- I think we should query the eth bridge status when validating vote extensions. this way, vote extensions issued when the bridge is inactive are rejected by `CheckTx`, but for this to work...
- we may need to refactor the way we store the activation status of the bridge. we need to query its status at a given epoch, the same when the vote extension was issued.
- it may be the case that a vote extension was issued during a period when the bridge was active, but then governance disabled the bridge before it got included, somehow? maybe this isn't an issue at all, because of pipeline lengths and so on

File: apps/src/lib/node/ledger/shell/vote_extensions.rs [hunk 2]
```diff
@@ -252,45 +265,59 @@ where
...
-        self.validate_eth_events_vext(
-            ext,
+        if !self.storage.is_bridge_active() {
+            ext.is_none()
+        } else {
+            if ext.is_none() {
+                return false;
+            }
+            self.validate_eth_events_vext(
+                ext.unwrap(),
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply apps/src/lib/node/ledger/shell/vote_extensions.rs:277]: I changed to an `if let` instead.

File: apps/src/lib/node/ledger/shell/vote_extensions.rs (hunk 1 above)
Comment (batconjurer) [reply apps/src/lib/node/ledger/shell/vote_extensions.rs:456]: That code isn't actually changed. It just registered as changed to github somehow.

**Review by sug0 — APPROVED** [review]
I can handle any validator set update code that is required to be changed in a later PR, so I'll approve this one

**Review by james-chf — APPROVED** [review]
`make check-abcipp` is failing in CI, otherwise lgtm

**Pull request merged**